### ISSUE TITLE
Generating multiple optimized sequences

### ISSUE THREAD

**Issue by zhxiaokang** [FPPGroup/CodonBERT#10]
I recently started studying the topic of codon optimization, and ran into your great work. But I have a question: is it possible for CodonBERT to generate multiple optimized sequences given one amino acid sequence?

Some other optimization tools can propose multiple candidates. For example, iCodon will propose 10 optimized sequences with different predicted stability, and CodonTransformer provides a parameter for users to decide the number of proposed candidates by themselves.

Can CodonBERT be modified somehow to also generate multiple candidates?

**Comment by renzilin** [comment]
Hi there，

CodonBERT can't generate multiple candidates. Because the model directly choose the codon with the highest probability. But you can modified the inference code to select the codon randomly according to the output probability.

**Comment by zhxiaokang** [comment]
Could you point me to the corresponding codes in the repo? Thank you!

**Comment by renzilin** [comment]
If you just want to use the model, you can modified `predict.py` in line 86-87.
```
annotation_logits_array = np.array(annotation_logits.detach().numpy())
result_annotation = np.argmax(annotation_logits_array, axis=1)
```
Here, we choose the codon whose index has the max logits. You can see it from the code above. And you can introduct a random function to generate the new index.
